model/task/local_service.go

```diff
@@ -29,18 +36,18 @@ func NewLocalService(env evergreen.Environment) *localService {
 
 // AppendTestResults appends test results to the local test results collection.
 func (s *localService) AppendTestResults(ctx context.Context, results []testresult.TestResult) error {
-	ids := map[testresult.DbTaskTestResultsID][]testresult.TestResult{}
+	infos := map[testresult.TestResultsInfo][]testresult.TestResult{}
 	for _, result := range results {
-		id := testresult.DbTaskTestResultsID{
+		info := testresult.TestResultsInfo{
 			TaskID:    result.TaskID,
 			Execution: result.Execution,
 		}
-		ids[id] = append(ids[id], result)
+		infos[info] = append(infos[info], result)
 	}
 
 	catcher := grip.NewBasicCatcher()
-	for id, results := range ids {
-		catcher.Add(s.appendResults(ctx, results, id))
+	for info := range infos {
+		catcher.Add(s.appendResults(ctx, results, info))
```

Comment:
Is this intentionally passing in all the test results, or is it supposed to pass just the test results for the one task ID/execution pair?

Reply:
The append test results function only ever invokes for a single task ID/execution pair so that's why I didn't notice this. Changed to assume only one task. However in the other PR #9129 this function will change to not behave this way anymore.